test/with_license_test.go

```diff
@@ -1,17 +1,93 @@
 package test
 
 import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strconv"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
+type MetricsItem []struct {
+	Name   string `json:"name"`
+	Metric Metric `json:"metric"`
+}
+
+type Metric []struct {
+	Gauge Gauge `json:"gauge"`
+}
+
+type Gauge struct {
+	Value int `json:"value"`
+}
+
+var (
+	client = &http.Client{}
+)
+
+func getNumberActiveQixSessions() int {
+	req, _ := http.NewRequest("GET", fmt.Sprintf("http://%s:9090/metrics", host), nil)
+	req.Header.Add("Accept", "application/json")
+	resp, _ := client.Do(req)
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	var metrics MetricsItem
+	json.Unmarshal(body, &metrics)
+
+	var activeSessions int
+	for _, metric := range metrics {
+		if metric.Name == "qix_active_sessions" {
+			activeSessions = metric.Metric[0].Gauge.Value
+		}
+	}
+	return activeSessions
+}
+
+func getLicensesMetrics() string {
+	req, _ := http.NewRequest("GET", fmt.Sprintf("http://%s:9200/metrics", host), nil)
+	resp, _ := client.Do(req)
+	body, _ := ioutil.ReadAll(resp.Body)
+	return string(body)
+}
+
+func getLicenseTimeConsumed() int {
+	licensesMetrics := getLicensesMetrics()
```

Comment:
fetch metrics once and then send it as a parameter to both `getLicenseTimeConsumed ` and `getLicenseTimeTotal`?